File: LeagueItemExplorerFront/src/app/models/champion-spell.ts
import { ChampionImage } from "./champion-image";
import { LevelTip } from "./level-tip";
import { SpellVar } from "./spell-var";

export interface ChampionSpell {
    id: string;
    name: string;
    description: string;
    tooltip: string;
    leveltip: LevelTip;
    maxrank: number;
    cooldown: number[];
    cooldownBurn: string;
    cost: number[];
    costBurn: string;
    datavalues: {[key: string]: number[]};
    effect: number[][];
    effectBurn: string[];
    vars: SpellVar[];
    costType: string;
    maxammo: string;
    range: number[];
    rangeBurn: string;
    image: ChampionImage;
    resource: string;
}

File: LeagueItemExplorerJAR/LeagueItemExplorerJAR/src/main/java/com/app/LeagueItemExplorerJAR/models/ChampionSpell.java
package com.app.LeagueItemExplorerJAR.models;

import lombok.Data;

import java.util.List;
import java.util.Map;
@Data
public class ChampionSpell {
    private String id;
    private String name;
    private String description;
    private String tooltip;
    private LevelTip leveltip;
    private int maxrank;
    private List<Integer> cooldown;
    private String cooldownBurn;
    private List<Integer> cost;
    private String costBurn;
    private Map<String, List<Double>> datavalues;
    private List<List<Integer>> effect;
    private List<String> effectBurn;
    private List<SpellVar> vars;
    private String costType;
    private String maxammo;
    private List<Integer> range;
    private String rangeBurn;
    private ChampionImage image;
    private String resource;
}
